src/D2L.CodeStyle.Analyzers/Immutability/MutabilityAuditor.cs

```diff
@@ -3,15 +3,24 @@
 
 namespace D2L.CodeStyle.Analyzers.Immutability {
 	internal sealed class MutabilityAuditor {
-		public static bool IsAudited( ISymbol symbol, out Location location ) {
+		public static bool IsAudited( ISymbol symbol, out Location location, out Diagnostic diagnostic ) {
 			AttributeData attr;
+			diagnostic = null;
+
+			// Check if there is a conflicting audit AND unaudit
+			if( ( Attributes.Statics.Audited.IsDefined( symbol ) || Attributes.Mutability.Audited.IsDefined( symbol ) )
+				&& ( Attributes.Statics.Unaudited.IsDefined( symbol ) || Attributes.Mutability.Unaudited.IsDefined( symbol ) ) ) {
+				diagnostic = Diagnostic.Create(
+					Diagnostics.ConflictingAuditing,
+					symbol.DeclaringSyntaxReferences[0].GetSyntax().GetLastToken().GetLocation() );
+			}
 
 			if( symbol.IsStatic ) {
 				attr = Attributes.Statics.Audited.GetAll( symbol ).FirstOrDefault()
-				    ?? Attributes.Statics.Unaudited.GetAll( symbol ).FirstOrDefault();
+					?? Attributes.Statics.Unaudited.GetAll( symbol ).FirstOrDefault();
 			} else {
 				attr = Attributes.Mutability.Audited.GetAll( symbol ).FirstOrDefault()
-				    ?? Attributes.Mutability.Unaudited.GetAll( symbol ).FirstOrDefault();
+					?? Attributes.Mutability.Unaudited.GetAll( symbol ).FirstOrDefault();

```

Comment:
We're still allowing

```cs
[Statics.Audited( ... )]
object x;
```

even though `x` is non-static, right? (this is mentioned in #661 as an undesirable thing)

Maybe we could do something like:

```cs
var hasStaticsAudited = ...;
var hasStaticsUnaudited = ...;
var hasBothStaticsAttributes = hasStaticsAudited  && hasStaticsUnaudited;
var hasEitherStaticsAttribute = hasStaticsAudited  || hasStaticsUnaudited;

var hasMutabilityAudited = ...;
var hasMutabilityUnaudited = ...;
var hasBothMutabilityAttribute = hasMutabilityAudited  && hasMutabilityUnaudited ;
var hasEitherMutabilityAttribute = hasMutabilityAudited  || hasMutabilityUnaudited ;

if( !hasEitherStaticsAttribute && !hasEitherMutabilityAttribute ) {
  // no attributes at all is ok
  return false;
}

if( hasBothStaticsAttributes ) {
  // diagnostic: can't have both audited and unaudited
  return true;
}

if( hasBothMutabilityAttributes ) {
  // diagnostic: can't have both audited and unaudited
  return true;
}

if ( symbol.IsStatic && hasEitherMutabilityAttribute ) {
  // diagnostic: wrong kind
  return true;
}

if ( !symbol.IsStatic && hasEitherStaticsAttribute ) {
  // diagnostic: wrong kind
  return true;
}

return true;
```

This would give you the ability to customize the diagnostic via `messageFormat` to the scenario.
You'd need an extra diagnostic for "wrong kind" (which would also need some `messageFormat`ing)